src/main/java/redis/clients/jedis/JedisCommandListener.java

```diff
@@ -0,0 +1,51 @@
+package redis.clients.jedis;
+
+import redis.clients.jedis.commands.ProtocolCommand;
+
+/**
+ * A listener for command events.
+ * <p>
+ * All commands on the same connections are executed synchronously, however there the
+ * implementation must be thread safe because the listener might be invoked from
+ * different threads.
+ * <p>
+ * The listener is guaranteed to call {@link #commandStarted} and one of the
+ * {@link #commandFinished} or {@link #commandFailed} once for each command.
+ * <p>
+ * The listener implementation must be exception free or else none of the aforementioned
+ * contracts will be enforced.
+ */
+public interface JedisCommandListener {
```

Comment:
Should extend EventListener?

Reply:
Hey, @gkorland! Do you mean `java.util.EventListener`? Out of interest, what is the benefit of extending `EventListener`? As far as I'm aware, it doesn't have much application outside of the swing/AWT world.

Reply:
I guess if we're not going with EventListenerList then we don't need it to be an EventListener

Reply:
Makes sense to me.  It's easy to implement this w/ a standard list and not include a swing dependency.  See my latest commit.